### PR TITLE
[MKLDNN] Develop fake data demo

### PR DESCRIPTION
**Pull request by lidanqing-vv**
We are working on speeding up multi-instance throughput caused by memory-bound issues. 
Keeping real data tests for verifying the accuracy, here I add fake data tests to easily provide reproduction steps for the issue and find more reducing memory opportunities.

| resnet50 int8 bs1 thread 1| develop:63f5c2d |
|:------------:|:------------------:|
| real data | 82.12 |
| fake data | 85.85 |

### CODE REVIEW

**Comment by lidanqing-vv** [comment]
@yeliang2258 Can you please review this PR? If we merge it, we don't have to always clone my repo for benchmarking SPR

**Comment by lidanqing-vv** [comment]
@jiangjiajun Could you please also approve this PR? Thank you!

File: demo/mkldnn_quant/sample_tester_fake_data.cc [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  input_t->copy_from_cpu<float>(input);
+  if (FLAGS_with_accuracy_layer){
+    int64_t* label = new int64_t[batch_size];
+    for (auto i=0;i<batch_size;i++) label[i]=batch_size%10+1;
+    auto input_l = predictor->GetInputTensor(input_names[1]);
+    input_l->Reshape({batch_size,1});
+    input_l->copy_from_cpu<int64_t>(label);
+  } 
+
+  for (auto iter = 0; iter<2; iter++){
```
Comment (yeliang2258) [inline demo/mkldnn_quant/sample_tester_fake_data.cc]: can use FLAGS_warmup_iter here

File: demo/mkldnn_quant/sample_tester.cc [hunk 2]
```diff
@@ -39,7 +39,7 @@ DEFINE_bool(with_accuracy_layer,
 DEFINE_bool(use_analysis,
             false,
             "If use_analysis is set to true, the model will be optimized");
-DEFINE_int32(warmup_iter, 2, "number of warmup batches");
```
Comment (yeliang2258) [inline demo/mkldnn_quant/sample_tester.cc]: Why do you need to change it to 1 ?

**Review by yeliang2258 — CHANGES_REQUESTED** [review]

File: demo/mkldnn_quant/sample_tester_fake_data.cc (hunk 1 above)
Comment (lidanqing-vv) [reply demo/mkldnn_quant/sample_tester_fake_data.cc]: Ok.

File: demo/mkldnn_quant/sample_tester.cc (hunk 2 above)
Comment (lidanqing-vv) [reply demo/mkldnn_quant/sample_tester.cc]: Updated. Please review again. Thanks!

File: demo/mkldnn_quant/sample_tester_fake_data.cc (hunk 1 above)
Comment (lidanqing-vv) [reply demo/mkldnn_quant/sample_tester_fake_data.cc]: Updated and tested. Please review again. Thanks!

**Review by yeliang2258 — DISMISSED** [review]
LGTM

**Review by yghstill — APPROVED** [review]

**Pull request merged**